### ISSUE TITLE
Proguard fails after upgrading to MapBox 5.3.1

### ISSUE THREAD

**Issue by stoefln** [mapbox/mapbox-gl-native#10960]
**Platform:**
Android (API 27), Gradle 4.1
**Mapbox SDK version:**
5.3.1

### Steps to trigger behavior

 1. Upgrade gradle plugin to 3.0.1
 2. Build release 
 3. Proguard fails with 54 warnings (all mapbox related)

### Gradle Console
```
...
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find superclass or interface com.mapzen.android.lost.api.LostApiClient$ConnectionCallbacks
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find superclass or interface com.mapzen.android.lost.api.LocationListener
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find superclass or interface com.mapzen.android.lost.api.LostApiClient$ConnectionCallbacks
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find superclass or interface com.mapzen.android.lost.api.LocationListener
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LostApiClient$Builder
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LostApiClient$Builder
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LostApiClient$Builder
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LostApiClient$Builder
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LostApiClient
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LostApiClient
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LocationServices
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.FusedLocationProviderApi
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LocationRequest
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LocationRequest
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LocationRequest
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LocationRequest
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LocationRequest
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LocationRequest
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.FusedLocationProviderApi
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.FusedLocationProviderApi
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LostApiClient
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LostApiClient$ConnectionCallbacks
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LocationListener
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LocationRequest
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LostApiClient
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LocationServices
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.FusedLocationProviderApi
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LostApiClient
Warning: com.mapbox.mapboxsdk.location.LocationSource: can't find referenced class com.mapzen.android.lost.api.LocationRequest
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LostApiClient$Builder
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LostApiClient$Builder
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LostApiClient$Builder
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LostApiClient$Builder
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LostApiClient
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LostApiClient
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LocationServices
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.FusedLocationProviderApi
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LocationRequest
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LocationRequest
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LocationRequest
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LocationRequest
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LocationRequest
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LocationRequest
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.FusedLocationProviderApi
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.FusedLocationProviderApi
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LostApiClient
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LostApiClient$ConnectionCallbacks
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LocationListener
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LocationRequest
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LostApiClient
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LocationServices
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.FusedLocationProviderApi
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LostApiClient
Warning: com.mapbox.services.android.telemetry.location.LostLocationEngine: can't find referenced class com.mapzen.android.lost.api.LocationRequest
...
```

**Comment by LukasPaczos** [comment]
Thanks for reaching out and reporting this problem @stoefln.
Are you using any of the [mapbox plugins](https://github.com/mapbox/mapbox-plugins-android) or adding `transitive` flag to the Mapbox Android SDK dependency in your gradle file?

**Comment by tobrun** [comment]
My guess is that this has to do making the location provider optional (concretely this means depending on lost with the `compileOnly/api` configuration in the build.gradle.

**Comment by stoefln** [comment]
@LukasPaczos yes, we use a plugin.  And we use the transitive flag, yes:
```
    implementation('com.mapbox.mapboxsdk:mapbox-android-sdk:5.3.1@aar') {
        transitive = true
    }
    implementation ('com.mapbox.mapboxsdk:mapbox-android-plugin-locationlayer:0.3.0') {
        exclude group: 'android.arch.core'
    }
```

**Comment by stoefln** [comment]
@tobrun not sure what you are saying...

**Comment by LukasPaczos** [comment]
@stoefln would you mind trying to remove `@aar` suffix and the transitive flag? It's not needed anymore.
Let us know whether that helps.

**Comment by stoefln** [comment]
@LukasPaczos unfortunately that didn't change anything...

**Comment by stoefln** [comment]
I didn't find anything in the documentation about setting proguard rules. still there is such a file here:
https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/MapboxGLAndroidSDK/proguard-rules.pro

what about it? Do I need to set something?

**Comment by stoefln** [comment]
adding "com.mapzen.android:lost:3.0.2" to dependencies solved my issue.
`gradlew :app:dependencies` did not show any dependency to lost though. Strange.

**Comment by Sarahhuang** [comment]
I solved this problem by adding `-dontwarn com.mapzen.android.lost.api**`

**Comment by accursedm0** [comment]
Same here with Mapbox SDK 5.4.0. 

Including the dependency or adding the dontwarn line solves the issue though. 
@stoefln @Sarahhuang Thanks for the hint!

**Comment by tobrun** [comment]
landed with #11127
